### PR TITLE
Poc dependencies ck merge

### LINKED ISSUE #18: Merge PoC changes into master

### PR DESCRIPTION
**Pull request by MikeCamel**
Fixes #18

Also moves to upstream ciborium

### CODE REVIEW

**Comment by MikeCamel** [comment]
@mbestavros - this is failing on the ring licence.  I believe that we fixed this elsewhere - please could we apply the same fix here?

**Comment by haraldh** [comment]
err.. these commits should be squashed and the text should have meaningful commit messages

File: Cargo.toml [hunk 1]
```diff
@@ -8,8 +8,14 @@ license = "Apache-2.0"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-sev = { git  = "https://github.com/enarx/sev" }
+sev = { git = "https://github.com/enarx/sev" }
 serde = { version = "1.0", features = ["derive"] }
 tokio = { version = "0.2", features = ["macros", "sync"] }
+#tokio = { version = "1.0", features = ["macros", "sync"] }
```
Comment (haraldh) [inline Cargo.toml:14]: with comment?

File: src/attestation/sev.rs [hunk 2]
```diff
@@ -6,6 +6,7 @@
 //!
 //! [1] https://github.com/enarx/enarx-keepldr/wiki/AMD-SEV-Remote-Attestation-Protocol
 
+//taken wholesale for https://github.com/connorkuehl/koine/commit/8659386bbdce554872231636d00a4b94c69f3aa2
```
Comment (haraldh) [inline src/attestation/sev.rs:9]: ?

File: src/lib.rs [hunk 3]
```diff
@@ -1,73 +1,106 @@
 // SPDX-License-Identifier: Apache-2.0
 
+use http::response::*;
 use serde::{Deserialize, Serialize};
-use std::sync::Arc;
-use tokio::sync::Mutex;
+use std::error::Error;
+use std::fmt;
+use std::path::PathBuf;
+//use std::net::IpAddr;
```
Comment (haraldh) [inline src/lib.rs]: ?

File: src/lib.rs [hunk 4]
```diff
@@ -1,73 +1,106 @@
...
+    //TODO - add further information
 }
-
+/*
+#[derive(Serialize, Deserialize, Clone)]
+pub struct Payload {
+    pub encoding: String,
+    pub contents: Vec<u8>,
+}
+*/
```
Comment (haraldh) [inline src/lib.rs]: ?

File: src/lib.rs [hunk 5]
```diff
@@ -93,6 +125,66 @@ pub struct UndefinedReply {
...
+}
+/*
+impl LocalCborErr {
+    fn new(msg: &str) -> LocalCborErr {
+        LocalCborErr {
+            details: msg.to_string(),
+        }
+    }
+}
+*/
```
Comment (haraldh) [inline src/lib.rs]: ?

File: src/lib.rs [hunk 6]
```diff
@@ -93,6 +125,66 @@ pub struct UndefinedReply {
...
+}
+
+//TODO - remove?
+pub trait MIMEPayload {
+    //NOTE -  all struct implementing this trait
+    // also need to derive cbor::Deserialize
+    // and cbor::Serialize
+    fn mime_type(&self) -> &'static str;
+}
+*/
```
Comment (haraldh) [inline src/lib.rs]: ?

File: src/threading/lists.rs [hunk 7]
```diff
@@ -0,0 +1,9 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use super::super::*;
```
Comment (haraldh) [inline src/threading/lists.rs:3]: ?

**Review by haraldh — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 4 above)
Comment (MikeCamel) [reply src/lib.rs]: fixed

File: src/lib.rs (hunk 5 above)
Comment (MikeCamel) [reply src/lib.rs]: fixed

File: src/lib.rs (hunk 6 above)
Comment (MikeCamel) [reply src/lib.rs]: fixed

File: src/lib.rs (hunk 3 above)
Comment (MikeCamel) [reply src/lib.rs]: fixed

File: Cargo.toml (hunk 1 above)
Comment (MikeCamel) [reply Cargo.toml:14]: fixed

File: src/threading/lists.rs (hunk 7 above)
Comment (MikeCamel) [reply src/threading/lists.rs:3]: removed this file

**Pull request closed without merging**